src/hooks/useResizer.js

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+
+import { addActions, actions } from "../actions";
+import { defaultState } from "./useTableState";
+
+defaultState.resizedColumns = {};
+
+addActions({
+  resizeColumn: "__resizeColumn__"
+});
+
+const propTypes = {
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      width: PropTypes.number
+    })
+  ),
+  onResizedChange: PropTypes.func
+};
+
+export const useResizer = props => {
+  PropTypes.checkPropTypes(propTypes, props, "property", "useResizer");
+
+  const {
+    columns,
+    hooks: { headers: headerHooks },
+    state: [{ resizedColumns }, setState],
+    getResizerProps,
+    onResizedChange
+  } = props;
+  const currentlyResizingInfo = useRef(undefined);
+
+  useEffect(() => {
+    if (resizedColumns) {
+      Object.keys(resizedColumns).forEach(index => (columns[index].width = resizedColumns[index]));
+    }
+  }, [resizedColumns]);
+
+  const addResizer = (columns, api) => {
+    let bottomLevelColumnCount = 0;
+
+    columns.forEach(column => {
+      if (!column.columns) {
+        const index = bottomLevelColumnCount;
+
+        column.isBottomLevel = true;
```

Comment:
Is the bottom level detection necessary? If it is, then I wonder how we could make hooks like this one easier to write in the future. Detecting bottom-level headers shouldn't be this difficult...

Reply:
It's  not  entirely necessary, just need to know what direction to go.  Could have it affect all columns including groups, which could be desired, or could affect only the bottom level.   You could argue that leaving it up to the user might be more in-line with your v7 philosphies.

Yea I was actually not super happy with this, but it worked for the time.